### ISSUE TITLE
Bref Serverless v4

### ISSUE THREAD

**Issue by cybage-kmistry** [brefphp/bref#1815]
**Description:**

Serverless comes up with version 4 which is licensed one and with bref during serverless deploy it is asking to login to it. 
How this will be handled with bref as bref is open source from start?

**How to reproduce:**

serverless deploy

**Comment by stloyd** [comment]
There was a recent adjustment to lock it to version 3: https://github.com/brefphp/bref/pull/1812

**Comment by austencollins** [comment]
fwiw the Serverless Framework's intention is to support Bref, as always. We've found a bug in V4 due to our new Variable Resolver logic, and we're working on a fix. We're tracking the issue [here](https://github.com/serverless/serverless/issues/12570).

**Comment by austencollins** [comment]
We've fixed support for the bug outlined in the issue above in [this release](https://github.com/serverless/serverless/releases/tag/v4.0.35). Bref should work seamlessly with Serverless Framework V4 as of now. We'll continue to test and monitor it. Please let us know if you see any further issues.

**Comment by Kremzeeq** [comment]
Hi, I can also report there's still an issue with login for the serverless framework.
Here's some code I can share from using `GitHub Actions:
```
    - name: serverless deploy
      run: | 
        npm i -g serverless@4.0.35
        npm install && npm list -g
        serverless deploy --verbose
      env:
        SERVERLESS_ACCESS_KEY: ${{ secrets.SERVERLESS_ACCESS_KEY }}
```

GitHub Actions logs:
```
Run npm i -g serverless@4.0.35

added 53 packages in 2s

15 packages are looking for funding
  run `npm fund` for details
npm warn deprecated inflight@1.0.6: This module is not supported, and leaks memory. Do not use it. Check out lru-cache if you want a good and tested way to coalesce async requests by a key value, which is much more comprehensive and powerful.
npm warn deprecated glob@7.2.3: Glob versions prior to v9 are no longer supported
npm warn deprecated querystring@0.2.1: The querystring API is considered Legacy. new code should use the URLSearchParams API instead.
npm warn deprecated superagent@7.1.6: Please upgrade to v9.0.0+ as we have fixed a public vulnerability with formidable dependency. Note that v9.0.0+ requires Node.js v14.18.0+. See https://github.com/ladjs/superagent/pull/1800 for insight. This project is supported and maintained by the team at Forward Email @ https://forwardemail.net
npm warn deprecated querystring@0.2.0: The querystring API is considered Legacy. new code should use the URLSearchParams API instead.
npm warn deprecated querystring@0.2.0: The querystring API is considered Legacy. new code should use the URLSearchParams API instead.

added 587 packages, and audited 588 packages in 14s

113 packages are looking for funding
  run `npm fund` for details

found 0 vulnerabilities
/opt/hostedtoolcache/node/[20](https://github.com/KremzeeqOrg/fruit-project-api-scraper/actions/runs/9563233246/job/26361335677#step:7:21).14.0/x64/lib
├── corepack@0.28.1
├── npm@10.7.0
└── serverless@4.0.35


Initializing

✖ Unauthorized
Error: Unauthorized
    at Authentication.getClientDataFromAccessKey (file:///home/runner/.serverless/releases/4.1.4/package/dist/sf-core.js:1060:27526)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async Authentication.getAuthenticatedData (file:///home/runner/.serverless/releases/4.1.4/package/dist/sf-core.js:1060:10955)
    at async Authentication.resolverAuthenticate (file:///home/runner/.serverless/releases/4.1.4/package/dist/sf-core.js:1060:31012)
    at async _Meta.authenticateAndResolve (file:///home/runner/.serverless/releases/4.1.4/package/dist/sf-core.js:1068:10639)
    at async route (file:///home/runner/.serverless/releases/4.1.4/package/dist/sf-core.js:1068:14642)
    at async Object.run2 [as run] (file:///home/runner/.serverless/releases/4.1.4/package/dist/sf-core.js:1068:15090)
    at async run3 (file:///home/runner/.serverless/releases/4.1.4/package/dist/sf-core.js:1068:163[24](https://github.com/KremzeeqOrg/fruit-project-api-scraper/actions/runs/9563233246/job/26361335677#step:7:25))

For help, try the following:
  • Run the command again with the "--debug" option
  • Run "serverless support"
  • Review the docs: https://www.serverless.com/framework/docs/

```
If you're curious about specifics:
-  https://github.com/KremzeeqOrg/fruit-project-api-scraper/pull/3/files
- https://github.com/KremzeeqOrg/fruit-project-api-scraper/actions/runs/9563233246/job/26361335677

I've closed my pr for now, but this basically covers a migration from serverless@3.2 to serverless@4.0.35. Deploy with serverless@3.2 was working before.

**Comment by austencollins** [comment]
Hmm, @Kremzeeq We have released multiple versions since `4.0.35` Could you try again please?

**Comment by austencollins** [comment]
Thanks, @Jimbolino. It looks like there are some unconventional require statements to fetch dependencies directly from the Framework, rather than go through the APIs the Framework provides for this functionality. We'll provide a PR shortly for this.

**Comment by Kremzeeq** [comment]
> Hmm, @Kremzeeq We have released multiple versions since `4.0.35` Could you try again please?

Hi @austencollins . I've now deployed the latest version - `4.1.7` and have tested this with a lambda. lgtm. Thanks you.

**Comment by austencollins** [comment]
Fix awaiting approval for the issue @Jimbolino mentioned: https://github.com/brefphp/bref/pull/1831

**Comment by mnapoli** [comment]
Closed by #1837

**Comment by amiretemad** [comment]
> im still having a small error with the latest version. probably something with this require: https://github.com/brefphp/bref/blob/master/index.js#L171
> 
> ```
> Finishing
> endpoint: ANY - [https://pml52gg7uc.execute-api.***.amazonaws.com](https://pml52gg7uc.execute-api.%2A%2A%2A.amazonaws.com/)
> functions:
>   web: troep-production-web (17 MB)
>   artisan: troep-production-artisan (17 MB)
> s:core:meta: saving meta record to global .serverless directory.
> s:cli-error-handler: Cannot read properties of undefined (reading 'require')
> TypeError: Cannot read properties of undefined (reading 'require')
>     at process.<anonymous> (/home/runner/work/www.troep.com/www.troep.com/vendor/bref/bref/index.js:171:44)
>     at process.emit (node:events:519:28)
>     at process.emit (node:domain:488:12)
> 
> ✖ Cannot read properties of undefined (reading 'require')
> TypeError: Cannot read properties of undefined (reading 'require')
>     at process.<anonymous> (/home/runner/work/www.troep.com/www.troep.com/vendor/bref/bref/index.js:171:44)
>     at process.emit (node:events:519:28)
>     at process.emit (node:domain:488:12)
> 
> For help, try the following:
>   • Run the command again with the "--debug" option
>   • Run "serverless support"
>   • Review the docs: https://www.serverless.com/framework/docs/
> 
> Error: Process completed with exit code 1.
> ```

I faced the same issue but managed to resolve it by updating Bref to version 2.3.0.
